backend/src/test/java/pl/edu/pw/elka/minedBlocks/MinedBlocksTests.java

```diff
@@ -0,0 +1,86 @@
+package pl.edu.pw.elka.minedBlocks;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import pl.edu.pw.elka.etherscan.EtherscanFacade;
+import pl.edu.pw.elka.minedBlocks.dtos.MinedBlockDto;
+import pl.edu.pw.elka.minedBlocks.dtos.MinedBlocksDto;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.mockito.Matchers.any;
+
+public class MinedBlocksTests {
+
+    private static final String MY_ADDRESS = "0xe103f7baef7a048cb22a4d566a37f72f762df229";
+
+    private MinedBlocksFacade minedBlocksFacade;
+    private EtherscanFacade etherscanFacade;
+
+    @Before
+    public void before() {
+        etherscanFacade = Mockito.mock(EtherscanFacade.class);
+        minedBlocksFacade = new MinedBlocksFacade(etherscanFacade);
+    }
+
+    @Test
+    public void shouldReturnNoBlocks() {
+        mockEmptyBlocks();
+        final MinedBlocksDto blocks = minedBlocksFacade.getMinedBlocksForAddress(MY_ADDRESS);
+        assertThat(blocks.getMinedBlocksRewards()).hasSize(0);
+    }
+
+    @Test
+    public void shouldReturnOneBlock() {
+        mockOneBlock();
+        final MinedBlocksDto blocks = minedBlocksFacade.getMinedBlocksForAddress(MY_ADDRESS);
+        assertThat(blocks.getMinedBlocksRewards()).hasSize(1);
+    }
+
+    @Test
+    public void shouldReturnTwoBlocks() {
+        mockTwoBlocks();
+        final MinedBlocksDto blocks = minedBlocksFacade.getMinedBlocksForAddress(MY_ADDRESS);
+        assertThat(blocks.getMinedBlocksRewards()).hasSize(2);
+    }
+
+    @Test
+    public void shouldReturnOneBlockWith1Ether() {
+        mockOneBlockWithValue("1");
+        final MinedBlocksDto blocks = minedBlocksFacade.getMinedBlocksForAddress(MY_ADDRESS);
+        MinedBlockDto block = blocks.getMinedBlocksRewards().stream().findAny().get();
+        assertThat(block.getBlockReward()).isEqualTo("1");
+    }
+
+    private void mockOneBlockWithValue(String value) {
+        final MinedBlockDto block = new MinedBlockDto(value);
+        final List<MinedBlockDto> blocks = Collections.singletonList(block);
+        Mockito.when(etherscanFacade.getMinedBlocksForAddress(any()))
+                .thenReturn(new MinedBlocksDto(blocks));
+    }
+
+    private void mockTwoBlocks() {
+        final MinedBlockDto[] blockDtos = {
+                new MinedBlockDto(null),
+                new MinedBlockDto(null)
+        };
+        Mockito.when(etherscanFacade.getMinedBlocksForAddress(any()))
+                .thenReturn(new MinedBlocksDto(Arrays.asList(blockDtos)));
+    }
+
+    private void mockOneBlock() {
+        final MinedBlockDto block = new MinedBlockDto(null);
+        final List<MinedBlockDto> blocks = Collections.singletonList(block);
+        Mockito.when(etherscanFacade.getMinedBlocksForAddress(any()))
+                .thenReturn(new MinedBlocksDto(blocks));
+    }
+
+    private void mockEmptyBlocks() {
```

Comment:
te testy z tego co widze sa dosc podobne do tych ktore juz sa - to jest ok, ale brakuje testow do logiki ktora znajduje liczy laczna wartosc wykopanych blokow - warto jak juz napisalem przeniesc ja do minedBlocks i dopisac pewnie wtedy testy - nie wiem czy te testy nie beda troche do wyrzucenia wtedy (a moze ich), bo zmieni sie api klasy MinedBlocksFacade (wydaje mi sie ze powinna byc metoda zwracajaca BigDecimala dla adresu, bo nic innego nas nie obodzi poza tym pakietem), warto tez dopisac jakis test w TransactionFacade z zamockowanym MinedBlocksFacade

Reply:
Przeniosłem logikę do MinedBlocks, jednak nie za bardzo wiem jak zrobić test dla zmockowanego MinedBlocksFacade w TransactionTests. Napisałem wstępnie dwa, jednak wykomentowałem je, gdyż z jakiegoś powodu oba zwracają null zamiast poprawnej wartości, tak jakby funkcja getMinedBlocksForAddress() z klasy MinedBlocksFacade nie stosowała się do Mockito.when w testach.